### PR TITLE
Adding relevant unit tests for MvcRazorCodeParser property name trimming

### PR DESCRIPTION
**Pull request by pranavkm**
cc @NTaylorMullen

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs [hunk 1]
```diff
@@ -177,87 +177,21 @@ public static TheoryData GlobalImportData
...
+        [InlineData("a bc d", "a bc d")]
+        [InlineData("abc;", "abc")]
+        [InlineData("  abc;", "abc")]
+        [InlineData(";; abc;", ";; abc")]
+        [InlineData("abc;; ;", "abc")]
+        public void RemoveWhitespaceAndTrailingSemicolons_ReturnsExpectedValues(string input, string expectedOutput)
         {
-            // Arrange & Act
-            var output = MvcRazorCodeParser.TrimSpacesAndChars(input, trimCharacters);
+            // Act
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: nit: still `// Arrange & Act`

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs [hunk 2]
```diff
@@ -177,87 +177,21 @@ public static TheoryData GlobalImportData
...
+        [InlineData("", "")]
+        [InlineData("   ;  ", "")]
+        [InlineData("    ", "")]
+        [InlineData("abcd", "abcd")]
+        [InlineData("abc;d", "abc;d")]
+        [InlineData("a bc d", "a bc d")]
+        [InlineData("abc;", "abc")]
+        [InlineData("  abc;", "abc")]
+        [InlineData(";; abc;", ";; abc")]
+        [InlineData("abc;; ;", "abc")]
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: suspect you have an off-by-one error in the product code.  example case would be `[InlineData(";;", "")]`.  method will instead return `";"`.  add case and fix bug.

Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: hmm, actually that case will work but `[InlineData("a", "a")]` will not.  method will instead return `string.Empty`.

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs (hunk 2 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: test other whitespace also. tabs, newlines etc.

**Comment by NTaylorMullen** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs (hunk 2 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: Tabs sound fine - but we aren't going to have newlines in `@inject` property name

**Comment by pranavkm** [comment]
Updated.

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs [hunk 3]
```diff
@@ -177,87 +177,33 @@ public static TheoryData GlobalImportData
...
         [Theory]
-        [MemberData(nameof(TrimSpacesAndCharsData))]
-        public void TrimSpacesAndChars_GeneratesExpectedOutput(
-            string input,
-            char[] trimCharacters,
-            string expectedOutput)
+        [InlineData("", "")]
+        [InlineData("   ;  ", "")]
+        [InlineData("    ", "")]
+        [InlineData("a", "a")]
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: Thanks.  Please add the `[InlineData(";;", "")]` case as well.

**Comment by dougbu** [comment]
:shipit: (with one more test case) once @NTaylorMullen is satisfied.

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**